sdk/core/src/bytes_response.rs

```diff
@@ -92,7 +92,7 @@ mod mock_transport {
     #[derive(Serialize, Deserialize)]
     pub(crate) struct SerializedBytesResponse {
         status: u16,
-        headers: HashMap<String, String>,
+        headers: BTreeMap<String, String>,
```

Comment:
What is the purpose of this? Perhaps removing this change and opening a new PR for it would be easier.

Reply:
> then arrange that serialisation for the mock transport framework puts headers in alphabetical order
>  - so that diffs like the one in the final commit have some chance of being reviewable

I put it in the same MR so that the third commit in the series would benefit from having it; but if this is controversial then we can do it separately.

Reply:
@rylev #475 is this MR without the alphabetical ordering, take your pick.